src/commandline_options.cpp

```diff
@@ -42,8 +40,8 @@ static void validate(boost::any& v, const std::vector<std::string>& values,
 	if (values.size() != 2) {
 		throw po::validation_error(po::validation_error::invalid_option_value);
 	}
-	ret_val.first = values.at(0);
-	ret_val.second = values.at(1);
+	ret_val.first = values[0];
+	ret_val.second = values[1];
```

Comment:
This file has other places where it uses [1] and [0]. I'll try to see and post a compiler explorer link. But we're double checking something that can't happen. Would it not be nicer to show the intent better and not re-check?

Maybe the compile could optimize it away, it may not do so in debug.

Reply:
you were 100% right! the compiler does manage to optimize the second if/exception throwing on all major platforms that wesnoth supports.

I'm a little surprised, even though I knew optimizers are insanely good

I expected that, at least in debug, it would emit more code for the .at(..)

here's the link: https://godbolt.org/z/r1x1P19vT

I'll revert it, but what happens with lines: 593, 616 (as of now) where the same pattern is used, i.e. check size != 2, throw, then used operator [] (not .at). do we want to change those to .at as well? do we change everything to []?

Reply:
done

Reply:
Thanks - I'll admit I'm flip-flopping on the decision considering that it's inconsistent with the other lines, and that all the code in this area is obviously doing input validation. Reading it as a file rather than just a diff, I'm currently thinking your change to using `[0]` and `[1]` here was the right one.

@CelticMinstrel do you have an opinion please?

Reply:
I generally default to using `[]` rather than `.at()` unless:

* You actually want the exception that `.at()` throws (perhaps you're planning to catch it) or
* It's a const map, so `[]` is not permitted (this point doesn't apply if it's a vector).